### DISCUSSION TITLE
⚠️ Avoid LilyGO devices with T-U2T ⚠️

### DISCUSSION THREAD

**Discussion by Jason2866** [arendst/Tasmota#16060] [General]
since the are not user friendly designed. See #16054 and other devices from them which uses T-U2T.
It is not norm compliant how T-U2T reuses the USB-C connector. This can end in defective devices and computers. Beside this you have to buy there highly priced USB-serial adapter T-U2T to have a easy way for flashing. Non of the needed gpios for flashing the ESP32 is broken out anywhere on the PCB.

There are better and user friendly alternatives. For [example](https://templates.blakadder.com/robotdyn_ESP32R4.html)

Tasmota nor any developer crew member of Tasmota are related to ANY manufacturer of devices!

**Comment by TD-er** [comment]
Related issue: https://github.com/Xinyuan-LilyGO/LilyGo-T-Relay/issues/19

In short, the battery charge specification demands some resistor between D+ and D- on the USB port.
On these T-U2T capable boards they re-use the USB-C connector and wire D+ and D- pins to RX and TX of the ESP.
Thus essentially everything you send out via TX will be echo'ed back to the RX pin.
